src/mpi/mpi_caf.c

```diff
@@ -8213,7 +8213,6 @@ void PREFIX(form_team) (int team_id, caf_team_t *team,
   MPI_Comm *current_comm = &CAF_COMM_WORLD;
```

Comment:
This is a bug, isn't it? Shouldn't `form team` create the new team set from the old team's communicator.

Reply:
It looks OK. MPI 3.1 lists the following arguments for MPI_Comm_split:

```
MPI_COMM_SPLIT(comm, color, key, newcomm)
 IN     comm                        communicator (handle)
 IN     color                       control of subset assignment (integer)
 IN     key                         control of rank assigment (integer)
 OUT    newcomm                     new communicator (handle)
```

For the "comm" argument, `*current_comm` (aka `CAF_COMM_WORLD`), which represents the current team, is used. For the last argument ("newcomm"), which stores the new communicator (team) that is stored in the FORM TEAM *team-variable* argument, `newcomm` is used. [and yes, copy & paste & resulting browser spell check revealed a typo in the MPI 3.1 standard: "assigment" -> "assignment". I'll notify a colleague who is on the MPI standards committee...]

Reply:
Ah, I guess I find the name `CAF_COMM_WORLD` to be misleading in this context. I assumed `CAF_COMM_WORLD` would always be the communicator copied from `MPI_COMM_WORLD`, as opposed to the communicator for the present team...

Reply:
To further elaborate: I'm concerned that a second `FORM TEAM` would partition the entire set of images, rather than the current team's set of images. But I can hunt through the source in a second to verify that `CAF_COMM_WORLD` is getting set to the communicator for the current team. (Although, if this is the case I wonder if it is a good design decision since we should still probably keep track of our copy of `MPI_COMM_WORLD` on each image.)

Reply:
I think the variable name CAF_COMM_WORLD predates teams support; it would be better to have it named something like CAF_COMM_CURRENT_TEAM.

The original duplicate communicator of MPI_COMM_WORLD (i.e., the 'original' CAF_COMM_WORLD) can be accessed, e.g., by searching backwards through the used_teams linked list.